src/DistributedLocking/Consul/LockProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Cascade\DistributedLocking\Consul;
+
+use Cascade\DistributedLocking\ILockProvider;
+use Cascade\DistributedLocking\ILockSessionProvider;
+use SensioLabs\Consul\Services\KV;
+
+class LockProvider implements ILockProvider
+{
+    /** @var KV */
+    private $consulKeyValue;
+
+    /** @var ILockSessionProvider */
+    private $sessionProvider;
+
+    /**
+     * @param ILockSessionProvider $sessionProvider
+     * @param KV $consulKeyValue
+     */
+    public function __construct(ILockSessionProvider $sessionProvider, KV $consulKeyValue)
+    {
+        $this->sessionProvider = $sessionProvider;
+        $this->consulKeyValue = $consulKeyValue;
+    }
+
+    /**
+     * @param string $lockName The name of the lock to acquire
+     *
+     * @return mixed An opaque handle for the lock, or false if the lock could not be acquired
+     */
+    public function lock($lockName)
+    {
+        $sessionId = $this->sessionProvider->getSessionId();
+
+        $result = (string)$this->consulKeyValue->put($lockName, '', ['acquire' => $sessionId])->getBody();
```

Comment:
The result from `getBody()` isn't a string by default -- it's a stream (which is handy when you want to do things other than just get the entire contents of the response) -- but here I just want the string value.